integration/spark/shared/src/main/java/io/openlineage/spark/agent/util/AwsUtils.java

```diff
@@ -32,7 +32,16 @@ public static Optional<String> getGlueArn(SparkConf sparkConf, Configuration had
                   getGlueCatalogId(sparkConf, hadoopConf)
                       .map(glueCatalogId -> "arn:aws:glue:" + region + ":" + glueCatalogId));
     } else {
-      return Optional.empty();
+      try {
+        return awsRegion()
+            .flatMap(
+                region ->
+                    getGlueCatalogId(sparkConf, hadoopConf)
+                        .map(glueCatalogId -> "arn:aws:glue:" + region + ":" + glueCatalogId));
+      } catch (Exception e) {
+        log.error("Failed to retrieve Glue", e);
+        return Optional.empty();
+      }
```

Comment:
Is it on purpose that we connect amazon s3 even if we don't have s3 enabled?
I'm trying to setup OpenLineage with S3 Transport (over the Datahub acryl-spark-lineage package which we used earlier), but my log is getting filled with this error specifically.
I tried setting the `hive.metastore.client.factory.class` but because of this change, even when I set it to different values, it's still trying to connect to Amazon s3.

Note: I am in a heavily firewalled environment so we will get a bunch of timeouts in our production environment, where locally I'm getting a bunch of 403's with invalid token as it detects the AWS credentials I am trying to use for the s3 transport.

Reply:
Are you able to find out which line or method of this class resulted in the requests that caused problem on your side? 

From what I see in the code, there is:
```
} else {
      log.debug("Fetching current account ID to use as the catalog ID.");
      return Optional.of(AwsAccountIdFetcher.getAccountId());
    }
```
which does a call to get `accountId` and perhaps OL integration could do better in persisting account Id after the first call. 

Please mind that setting `spark.glue.accountId` could be a workaround to avoid this call.

Reply:
From what I remember, it's indeed the fetching of this ID. Then because I'm setting up authentication in a generic way, this call also uses those credentials and hence the 403.

If I set the value you mentioned, would it avoid doing any S3 calls, as in, the code after also wouldn't execute any S3 calls?

Reply:
The class `software.amazon.awssdk.services.sts.StsClient` is used only within `AwsAccountIdFetcher`. Method `io.openlineage.spark.agent.util.AwsAccountIdFetcher#getAccountId` is called only when no `hive.metastore.glue.catalogid` nor `spark.glue.accountId` provided. So, this should help. 

Alternative approach would be to extend `io.openlineage.spark.agent.util.AwsAccountIdFetcher` to configure authentication.